mainnet-contracts/src/PufferProtocol.sol

```diff
@@ -715,106 +886,195 @@ contract PufferProtocol is IPufferProtocol, AccessManagedUpgradeable, UUPSUpgrad
+        bytes[] calldata vtConsumptionSignature
+    ) internal returns (uint256 vtAmountToBurn) {
+        // Burn the VT first, then fallback to ETH from the node operator
+        uint256 nodeVTBalance = $.nodeOperatorInfo[nodeOperator].deprecated_vtBalance;
+
+        uint256 vtBurnAmount = _getVTBurnAmount($, nodeOperator, totalEpochsValidated);
+
+        // If the node operator has VT, we burn it first
+        if (nodeVTBalance > 0) {
+            if (nodeVTBalance >= vtBurnAmount) {
+                // Burn the VT first, and update the node operator VT balance
+                vtAmountToBurn = vtBurnAmount;
+                // nosemgrep basic-arithmetic-underflow
+                $.nodeOperatorInfo[nodeOperator].deprecated_vtBalance -= SafeCast.toUint96(vtBurnAmount);
+
+                emit ValidationTimeConsumed({ node: nodeOperator, consumedAmount: 0, deprecated_burntVTs: vtBurnAmount });
+
+                return vtAmountToBurn;
+            }
 
-        module.callStake({ pubKey: validatorPubKey, signature: validatorSignature, depositDataRoot: depositDataRoot });
+            // If the node operator has less VT than the amount to burn, we burn all of it, and we use the validation time
+            vtAmountToBurn = nodeVTBalance;
+            // nosemgrep basic-arithmetic-underflow
+            $.nodeOperatorInfo[nodeOperator].deprecated_vtBalance -= SafeCast.toUint96(nodeVTBalance);
+
+            _settleVTAccounting({
+                $: $,
+                node: nodeOperator,
+                totalEpochsValidated: totalEpochsValidated,
+                vtConsumptionSignature: vtConsumptionSignature,
+                deprecated_burntVTs: nodeVTBalance
+            });
+
+            return vtAmountToBurn;
+        }
+
+        // If the node operator has no VT, we use the validation time
+        _settleVTAccounting({
+            $: $,
+            node: nodeOperator,
+            totalEpochsValidated: totalEpochsValidated,
+            vtConsumptionSignature: vtConsumptionSignature,
+            deprecated_burntVTs: 0
+        });
     }
 
-    function _getVTBurnAmount(ProtocolStorage storage $, address node, StoppedValidatorInfo calldata validatorInfo)
+    function _settleVTAccounting(
+        ProtocolStorage storage $,
+        address node,
+        uint256 totalEpochsValidated,
+        bytes[] calldata vtConsumptionSignature,
+        uint256 deprecated_burntVTs
+    ) internal {
+        // There is nothing to settle if this is the first validator for the node operator
+        if ($.nodeOperatorInfo[node].activeValidatorCount + $.nodeOperatorInfo[node].pendingValidatorCount == 0) {
+            return;
+        }
+
+        // We have no way of getting the present consumed amount for the other validators on-chain, so we use Puffer Backend service to get that amount and a signature from the service
+        bytes32 messageHash = keccak256(abi.encode(node, totalEpochsValidated, _useNonce(node)));
```

Comment:
Do we need some sort of deadline here? to check if the signature is old. Is there any scenario where an old signature could be used to game the system?